Modules/arraymodule.c

```diff
@@ -3053,13 +3059,14 @@ array_modexec(PyObject *m)
         *p++ = (char)descr->typecode;
     }
     typecodes = PyUnicode_DecodeASCII(buffer, p - buffer, NULL);
-
-    PyModule_AddObject(m, "typecodes", typecodes);
-
-    if (PyErr_Occurred()) {
-        Py_DECREF(m);
-        m = NULL;
+    if (typecodes == NULL) {
```

Comment:
It is not needed. `PyModule_AddObject()` checks the added object for NULL.

If you remove it, change also `Py_DECREF(typecodes)` to `Py_XDECREF(typecodes)`.

Reply:
done in 777fc61cd2a06c17055ed6d2c7e59069f75c1223